### PR TITLE
Optimize constructors for `LogicalSexp` and `StringSexp`

### PR DESCRIPTION
**Pull request by daniellga**
Remove some bounds check on `LogicalSexp` and `StringSexp` constructors. I didn't mess with the iterator part since I don't know if you are finished there yet, but `set_elt_unchecked` can probably be used there as well.
I also removed a nested unsafe in one of your functions.
I can't manage to test your crate. I don't know how to... But it compiles.

### CODE REVIEW

File: src/sexp/string.rs [hunk 1]
```diff
@@ -180,21 +183,19 @@ impl OwnedStringSexp {
...
-            Ok(savvy_ffi::R_NaString)
-        } else {
-            crate::unwind_protect(|| {
-                Rf_mkCharLenCE(
-                    v.as_ptr() as *const c_char,
-                    v.len() as i32,
-                    cetype_t_CE_UTF8,
-                )
-            })
-        }
```
Comment (yutannihilation) [inline src/sexp/string.rs:197]: This is intentional in order to be ready for edition 2024. See their blog post

https://blog.rust-lang.org/inside-rust/2024/03/22/2024-edition-update.html
https://doc.rust-lang.org/nightly/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html

**Review by yutannihilation — APPROVED** [review]
Thanks! Basically, looks good to me. I want to add a few tweaks, but I'll do it on my side after merging.

**Comment by yutannihilation** [comment]
One question. You set these functions `pub(crate)`. Why? Do you have any plan to use it other places?

**Comment by daniellga** [comment]
> One question. You set these functions `pub(crate)`. Why? Do you have any plan to use it other places?

I just let it open in case they are used in another place... I have no strong opinion on this, do you want me to make them not public at all?

**Comment by yutannihilation** [comment]
I too have not strong opinion. I'm fine even if it's made visible to the users. But, as I don't see any possible usage, let me hide it for now. If you find a valid use case, please let me know!

Anyway, I made some tweaks here. I'm merging, but feel free to leave comments or questions.

https://github.com/yutannihilation/savvy/pull/146

**Pull request merged**